packages/@okta/vuepress-site/docs/guides/build-custom-ui-mobile/main/index.md

```diff
@@ -0,0 +1,75 @@
+---
+title: Build a custom sign-in UI in your mobile app
+excerpt: Learn how to build a custom sign-in UI in your mobile app.
+layout: Guides
+---
+
+You can connect your mobile app to Okta and sign users in by [opening a browser](/docs/guides/sign-into-mobile-app/), however, if you prefer that your users not leave your app, you need to build a custom sign-in UI with native controls and screens instead. Use this guide to build a customized sign-in experience inside your mobile application.
+
+> **Note:** If the browser sign-in method works for your application, we recommend using that since building a custom sign-in UI takes more effort and development time.
+
+---
+
+**Learning outcomes**
+
+Build a custom mobile sign-in UI
+
+**What you need**
+
+* An Okta Developer organization. Don't have one? [Create one for free](https://developer.okta.com/signup)
+* A mobile app with Okta authentication (see [Sign users in to your mobile app](/docs/guides/sign-into-mobile-app/)) that you want to add custom the sign-in UI to
+
+**Sample code**
+
+See the [Sample code](#sample-code) section.
+
+---
+
+## Create an Okta application
+
+Before you can sign a user in, you need to create an Okta application that represents your mobile application.
+
+Use the Okta application that you created when you walked through the [Sign users in to your mobile app](/docs/guides/sign-into-mobile-app/-/main/#create-an-okta-app-integration) guide.
+
+## Add and configure packages
+
+To build the custom sign-in UI, you need to install and configure a native Okta SDK to your application.
+
+You should already have added and configured packages when you walked through the [Sign users in to your mobile app](/docs/guides/sign-into-mobile-app/-/main/#add-and-configure-packages) guide.
+
+In addition, you need to install the native Okta Authentication SDK. This SDK works together with the OpenID Connect SDK that you already installed to make authentication requests to Okta.
+
+<StackSelector snippet="installoktaauthsdk" />
+
+## Build the primary authentication form
+
+Okta's Authentication SDK is built around a [state machine](/docs/reference/api/authn/#transaction-state). Review the available states before you use this library.
+
+You can implement an authentication flow using one screen or using multiple screens. When you use multiple screens, you can spilt responsibilties across screens and then inject related data as a dependency.
+
+For example, multiple screens could handle:
+
+* login/password input
+* multifactor enrollment
+* factor selection
+* multifactor verification
+
+<StackSelector snippet="primaryauth" />
```

Comment:
Are we keeping stack selector drop-down menus for each section?

Or should we use `inline` or `noSelector` options in the tags?

`<StackSnippet snippet="inlineSnippet" inline />`
`<StackSelector snippet="sample" noSelector />`